### DISCUSSION TITLE
Remotes causes Host to crash when not loaded in Module Federation

### DISCUSSION THREAD

**Discussion by ilyasTraikia** [webpack/webpack#17947] [Q&A]
@ramesh-murugan-at-philips 

As @ScriptedAlchemy said, you can use the advanced api for dynamic remotes to have more control over script loading.

I adapted / simplified the following  from the [dynamic-remotes example](https://github.com/module-federation/module-federation-examples/blob/master/advanced-api/dynamic-remotes/app1/src/App.js)
```js
const loadScope = (url, scope) => {
  const element = document.createElement('script');
  const promise = new Promise((resolve, reject) => {
    element.src = url
    element.type = 'text/javascript'
    element.async = true
    element.onload = () => resolve(window[scope])
    element.onerror = reject
  })
  document.head.appendChild(element)
  promise.finally(() => document.head.removeChild(element))
  return promise
}

const loadModule = async (url, scope, module) => {
  try {
    const container = await loadScope(url, scope)
    await __webpack_init_sharing__('default')
    await container.init(__webpack_share_scopes__.default)
    const factory = await container.get(module)
    return factory()
  } catch (error) {
    console.error('Error loading module:', error)
    throw error
  }
}

const RemoteButton = React.lazy(() => loadModule(
  'http://localhost:3002/remoteEntry.js',
  'app2',
  './Button'
))
```
Try using it in the basic-host-remote example (app1/src/App.js). It catches the ChunkLoadError I was having issues with.

_I still don't think we should have to drop down to the low level api to get proper error handling though._

_Originally posted by @Rinaldo in https://github.com/webpack/webpack/issues/11944#issuecomment-747082733_

  Well i dont know if you remember this issue  but  rn iam using module federation-mf for a nextjs app and the host app crashes when the remotes are not available , i tried your piece of code and it did not work, what do you suggest?

**Comment by ilyasTraikia** [comment]
@ScriptedAlchemy

**Comment by ScriptedAlchemy** [comment]
Check updates examples now on federation

**Comment by ilyasTraikia** [comment]
> Check updates examples now on federation

Which one exactly ? iam relying on the nextjs one , latest version.

**Comment by ilyasTraikia** [comment]
and also one last thing , i have a navbar but when navigating it does a hard reload , it only soft navigates when the remotes are off , is there a way i can go back to client navigation becasue thats a bad ux..i ll appreciate your help.
